java/cuvs-java/src/main/java22/com/nvidia/cuvs/internal/common/Native.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.nvidia.cuvs.internal.common;
+
+import static com.nvidia.cuvs.internal.panama.headers_h.*;
+
+import java.lang.foreign.*;
+import java.lang.invoke.MethodHandle;
+
+public final class Native {
+
+  private Native() {}
+
+  private static final Linker LINKER = Linker.nativeLinker();
+
+  private static final MethodHandle cudaMemcpyAsync$mh =
+      LINKER.downcallHandle(
+          cudaMemcpyAsync$address(), cudaMemcpyAsync$descriptor(), Linker.Option.critical(true));
+
+  /**
+   * Bindings for {@code cudaMemcpyAsync}; differently from the {@code headers_h} bindings (which are
+   * automatically generated by {@code jextract}), these bindings specify the {@code critical} linker option,
+   * which allows to directly access heap-based {@link MemorySegment}s.
+   * {@snippet lang=c :
+   * extern cudaError_t cudaMemcpyAsync(void *dst, const void *src, size_t count, enum cudaMemcpyKind kind, cudaStream_t stream)
+   * }
+   */
+  public static int cudaMemcpyAsync(
+      MemorySegment dst, MemorySegment src, long count, int kind, MemorySegment stream) {
```

Comment:
Nitpick: I'm not keen on the `int kind` parameter to this function.  `Util.cudaMemcpy` is a wrapper that takes a `CudaMemcpyKind`, an enum for the direction.

We could consider either moving that function to `Native`, or this function to `Util`, and have them both use `CudaMemcpyKind`.

Reply:
I was debating this myself -- it makes little sense to have both this here and `Util`. I will move everything to `Util`.

Reply:
(side note: the `int kind` parameter is kind of needed here, as it needs to match the underlying C binding data type. Moving this to Util will help disguise that)